src/ui/public/timefilter/get_time.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+import dateMath from '@kbn/datemath';
+
+export function calculateBounds(timeRange, options = {}) {
+  return {
+    min: dateMath.parse(timeRange.from, { forceNow: options.forceNow }),
+    max: dateMath.parse(timeRange.to, { roundUp: true, forceNow: options.forceNow })
+  };
+}
+
+export function getTime(indexPattern, timeRange, forceNow) {
+  if (!indexPattern) {
+    //in CI, we sometimes seem to fail here.
```

Comment:
😬 Can you justify this a little better please? Not sure what this means by "sometimes seem to fail", but if you mean that some tests aren't properly stubbing indexPattern and that's causing issues I think we should update the indexPattern stubs, or stub out this method in tests, or something other than return `undefined` if an index patter isn't passed. This type of silent failing has a tendency to bite back in weird ways down the road.

Reply:
no idea, just extracted this to its own file, i didn't write this code :)